### PR TITLE
Add LoginFile class + dependencies

### PR DESCRIPTION
**Pull request by sisou**
Since I was experimenting with the old-style backup file in the beginning, this PR also includes some improvements to the Iqons class.

~~Also includes @danimoh's QRScanner improvements that enable scanning an inverted QRCode.~~

This PR does not include changes required to encode the new Imagewallet standard. This will happen in a separate PR (#155).

### CODE REVIEW

File: tests/lib/LoginFile.spec.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+/* global Dummy */
+/* global LoginFile */
+/* global QrScanner */
+
+describe('LoginFile', () => {
+
+    it('can generate a Wallet File', async () => {
+        const entropy = new Nimiq.Entropy(Dummy.keys[0]);
+
+        // @ts-ignore
```
Comment (nibhar) [inline tests/lib/LoginFile.spec.js]: Please add what was ignored according to #168

File: demos/LoginFile.html [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width,initial-scale=1.0">
+    <meta http-equiv="Content-Security-Policy"
+          content="default-src 'self' 'unsafe-eval' 'unsafe-inline' https://cdn.nimiq-testnet.com;
+          img-src http: https: blob: data:;
+          child-src 'self' blob:;
+          style-src 'self' 'unsafe-inline' https://fonts.googleapis.com;
+          font-src https://fonts.gstatic.com;">
+    <title>LoginFile/Imagewallet | Nimiq Keyguard</title>
```
[the remark this answers is not in the document]
Comment (sisou) [reply demos/LoginFile.html]: It's a demo site, for developers only :P

File: src/components/FileImport.js [hunk 3]
```diff
@@ -72,17 +73,7 @@ class FileImport extends Nimiq.Observable {
...
-        const qrPosition = {
-            x: 156,
-            y: 548.6886,
-            width: 173.4,
-            height: 173.4,
-            size: 185.4,
-            padding: 12,
-        };
-
+        const qrPosition = LoginFile.calculateQrPosition();
```
[the remark this answers is not in the document]
Comment (sisou) [reply src/components/FileImport.js]: The `qrPosition` calculated here is the first search area. When no QR code is found in this area, the scanner scans the rest of the image.

[the remark this answers is not in the document]
Comment (sisou) [reply src/components/FileImport.js]: The last parameter here (https://github.com/nimiq/keyguard-next/pull/154/files/819e7c08799706006addbc76abe4580cc634858b#diff-8d1422cdec1b5339d293e64be2e487ecR78) is the `alsoTryWithoutSourceRect` param and is set to true.

File: src/lib/Iqons.js [hunk 4]
```diff
@@ -145,20 +145,24 @@ class Iqons {
      * @returns {Promise<Document>}
      */
     static async _getAssets() {
-        /** @type {Promise<Document>} */
-        this._assetPromise = this._assetPromise || fetch(this.svgPath)
+        // eslint-disable-next-line no-return-assign
+        return this._assetsPromise || (this._assetsPromise = fetch(window.NIMIQ_IQONS_SVG_PATH || Iqons.SVG_PATH)
```
[the remark this answers is not in the document]
Comment (sisou) [reply src/lib/Iqons.js]: It's a global parameter, much like an environment variable, to be set by the user of the library. I agree, that it doesn't make too much sense to have in the Keyguard, where we control the environment ourselves.

File: src/lib/LoginFile.js [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+            padding: LoginFile.QR_PADDING,
+            width: LoginFile.QR_BOX_SIZE,
+            height: LoginFile.QR_BOX_SIZE,
+        };
+    }
+
+    filename() {
+        const walletName = 'Nimiq Login File '
+                         + `${this._config.name[0].toUpperCase()}${this._config.name.substr(1)} Wallet`;
+        return `${walletName.replace(/ /g, '-')}.png`;
```
[the remark this answers is not in the document]
Comment (sisou) [reply src/lib/LoginFile.js]: Yes. I believe that the automatic address naming plays a role here? We need to discuss this with design, but also, multiple wallets might be the exception...

File: src/lib/LoginFile.js [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+            LoginFile.BORDER_WIDTH,
+            this._width - LoginFile.BORDER_WIDTH * 2,
+            this._height - LoginFile.BORDER_WIDTH * 2,
+            LoginFile.RADIUS,
+            true, false, true,
+        );
+    }
+
+    async _drawDecorations() {
+        const img1 = new Image();
```
[the remark this answers is not in the document]
Comment (sisou) [reply src/lib/LoginFile.js]: YES!!! Awesome suggestion! :+1: :tada:

File: src/lib/LoginFile.js [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+        const ctx = this._ctx;
+        const x = LoginFile.WIDTH / 2;
+        const y = 97;
+        const date = new Date();
+        /**
+         * @param {number} num
+         * @returns {string}
+         */
+        const leftPad = num => `${num < 10 ? '0' : ''}${num}`;
+        const datestring = `${date.getFullYear()}-${leftPad(date.getMonth() + 1)}-${leftPad(date.getDate())}`;
```
[the remark this answers is not in the document]
Comment (sisou) [reply src/lib/LoginFile.js]: `toISOString()` is always in UTC time, though, which we don't want. Thus the left-pad method is the best currently.

File: src/lib/Iqons.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (sisou) [reply src/lib/Iqons.js]: But its required by the demos, thus I'd just leave it in.

File: src/lib/LoginFile.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (nibhar) [reply src/lib/LoginFile.js]: Yes, I think the new naming scheme for accounts would come in handy here as it sort of serves as an textual identifier, which then should be used in my opinion to identify the files, too.
We can discuss that later, once we have that new scheme.

File: src/lib/Iqons.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (nibhar) [reply src/lib/Iqons.js]: At some point I'd like to think about how to get rid of it, as it does not serve a purpose beyond enabling a demo.

File: src/components/FileImport.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (nibhar) [reply src/components/FileImport.js]: I see, thanks for the clarification.

File: src/lib/LoginFile.js [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+                resolve(url);
+            });
+        });
+    }
+
+    /**
+     * @param {string} encodedPrivKey
+     * @returns {Promise<void>}
+     */
+    async _draw(encodedPrivKey) {
```
Comment (nibhar) [inline src/lib/LoginFile.js]:
```suggestion
    async _draw(encodedPrivateKey) {
```
no need to shorten the name.

File: src/lib/LoginFile.js [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+        await this._drawDecorations();
+        await this._drawNimiqLogo();
+
+        this._setFont();
+        this._drawDateText();
+        this._drawWarningText();
+
+        await this._drawIWLogo();
+
+        this._drawQrCode(encodedPrivKey);
```
Comment (nibhar) [inline src/lib/LoginFile.js]:
```suggestion
        this._drawQrCode(encodedPrivateKey);
```

File: src/lib/LoginFile.js [hunk 10]
```diff
@@ -0,0 +1,274 @@
...
+        this._ctx.drawImage(img, x, y, w, h);
+    }
+
+    /**
+     * @param {number} x
+     * @param {number} y
+     * @param {number} width
+     * @param {number} height
+     * @param {number} radius
+     * @param {boolean} [fill]
```
Comment (nibhar) [inline src/lib/LoginFile.js]:
```suggestion
     * @param {boolean} [fill = false]
```

File: src/lib/LoginFile.js [hunk 11]
```diff
@@ -0,0 +1,274 @@
...
+     * @param {boolean} [stroke]
```
Comment (nibhar) [inline src/lib/LoginFile.js]:
```suggestion
     * @param {boolean} [stroke = false]
```

File: src/lib/LoginFile.js [hunk 12]
```diff
@@ -0,0 +1,274 @@
...
+     * @param {boolean} [withIWCorner]
```
Comment (nibhar) [inline src/lib/LoginFile.js]:
```suggestion
     * @param {boolean} [withIWCorner = false]
```

File: src/lib/LoginFile.js [hunk 13]
```diff
@@ -0,0 +1,274 @@
...
+        await loaded;
+        this._ctx.drawImage(img, x, y, w, h);
+    }
+
+    /**
+     * @param {number} x
+     * @param {number} y
+     * @param {number} width
+     * @param {number} height
+     * @param {number} radius
```
Comment (nibhar) [inline src/lib/LoginFile.js]:
```suggestion
     * @param {number} [radius = 5]
```

File: src/lib/LoginFile.js [hunk 14]
```diff
@@ -0,0 +1,274 @@
+/* global QrEncoder */
+
+class LoginFile {
+    /**
+     * @param {string} encodedPrivKey
+     * @param {number} [color]
```
Comment (nibhar) [inline src/lib/LoginFile.js]:
```suggestion
     * @param {number} [color = 0]
```

**Pull request merged**